artifactory/services/utils/searchutil.go

```diff
@@ -74,10 +74,20 @@ func SearchBySpecWithBuild(specFile *CommonParams, flags CommonConf) (*content.C
 
 	wg.Wait()
 	if artifactsReader != nil {
-		defer artifactsReader.Close()
+		defer func() {
+			e := artifactsReader.Close()
+			if err == nil {
+				err = e
+			}
```

Comment:
Let's simplify with `errors.Join`.
This applies to multiple places in the code of this PR.

Reply:
Hi @sarao1310 and @eyalbe4,
Just wanted you to notice that all these changes are not doing anything.
The err = e functionality inside defers is working only on named return values.
The above code and also all the other ones here won't change the returned error.
It will work only with (**reader** *content.ContentReader, **err** error) and not with (*content.ContentReader, error).

Reply:
Absolutely correct @sverdlov93!

Reply:
I solved this issue using errors.join()